dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/feature/FeatureSdkCore.kt

```diff
@@ -71,4 +73,14 @@ interface FeatureSdkCore : SdkCore {
      * @param featureName Feature name.
      */
     fun removeEventReceiver(featureName: String)
+
+    /**
+     * Returns a new [ExecutorService], set up with backpressure and internal monitoring.
+     */
+    fun createExecutorService(): ExecutorService
+
+    /**
+     * Returns a new [ScheduledExecutorService], set up with internal monitoring.
+     */
+    fun createScheduledExecutorService(): ScheduledExecutorService
```

Comment:
I think it is a bit dangerous to not know how many threads will be managed under the hood by the service created.

We may have some change in the implementation of those methods in the future, for example `createExecutorService` won't return single-threaded service anymore, and we will face bugs which are hard to catch (because, for example, in unit tests we mock executors, so unlikely we will catch them there).

Maybe we can adopt lingvo from `Executors` class, which have methods like `newSingleThreadExecutor` (we may name `createExecutorService` exactly like this btw). And for the `createScheduledExecutorService` write in the javadoc that it will create an executor with max number of threads matching number of cores?